### ISSUE TITLE
encryypt-apk.py 

### ISSUE THREAD

**Issue by ketasic** [cryptax/angeapk#2]
Hi
after padding the file, i tried to encrypt it using encrypt-apk.py (the script that was generated in the beginning from running angecrypt. However as i run the script on the file it gives an error 
""  Traceback (most recent call last):
  File "encrypt-apk.py", line 7, in <module>
    d = algo.encrypt(d)
  File "/usr/lib/python2.7/dist-packages/Crypto/Cipher/blockalgo.py", line 244, in encrypt
    return self._cipher.encrypt(plaintext)
ValueError: Input strings must be a multiple of 16 in length
""
im a bit new to this and this may seem like a dumb problem but ive spent over a week now trying to figure it out.
hoping for a response.

**Comment by cryptax** [comment]
AES operates on 16 byte blocks. Your input must therefore be a multiple of 16 in length. If not, basically, this means you need padding.

**Comment by ketasic** [comment]
HI 
Yeah. IT helped. iT's working perfectly thank you.
I've got a favour to ask. if possible could you help with the wrapping-apk java script as well.  Whenever i try to compile it using javac command it produces the following errors
"
PocActivity.java:3: error: package android.widget does not exist
import android.widget.TextView;
                     ^
PocActivity.java:4: error: package android.widget does not exist
import android.widget.Button;
                     ^
PocActivity.java:5: error: package android.view does not exist
import android.view.View;
                   ^
PocActivity.java:6: error: package android.app does not exist
import android.app.Activity;
                  ^
PocActivity.java:7: error: package android.os does not exist
import android.os.Bundle;
                 ^
PocActivity.java:8: error: package android.content.res does not exist
import android.content.res.AssetManager;
                          ^
PocActivity.java:10: error: package android.content does not exist
import android.content.Intent; 
                      ^
PocActivity.java:11: error: package android.net does not exist
import android.net.Uri;
                  ^
PocActivity.java:17: error: package android.widget does not exist
import android.widget.Toast;
                     ^
PocActivity.java:18: error: package android.util does not exist
import android.util.Log;
                   ^
PocActivity.java:19: error: package android.content does not exist
import android.content.Context;
                      ^
PocActivity.java:20: error: package android.os does not exist
import android.os.Environment;
                 ^
PocActivity.java:21: error: package android.widget does not exist
import android.widget.ImageView;
                     ^
PocActivity.java:22: error: package android.graphics.drawable does not exist
import android.graphics.drawable.Drawable;
                                ^
PocActivity.java:24: error: cannot find symbol
public class PocActivity extends Activity
                                 ^
  symbol: class Activity
PocActivity.java:26: error: cannot find symbol
    public TextView txtView = null;
           ^
  symbol:   class TextView
  location: class PocActivity
PocActivity.java:27: error: cannot find symbol
    public ImageView mImage = null;
           ^
  symbol:   class ImageView
  location: class PocActivity
PocActivity.java:34: error: cannot find symbol
    public void onCreate(Bundle savedInstanceState)    {
                         ^
  symbol:   class Bundle
  location: class PocActivity
PocActivity.java:30: error: cannot find symbol
    public static final String PAYLOAD_APK = Environment.getExternalStorageDirectory() + "/hidden.apk";
                                             ^
  symbol:   variable Environment
  location: class PocActivity
PocActivity.java:33: error: method does not override or implement a method from a supertype
    @Override
    ^
PocActivity.java:35: error: cannot find symbol
        super.onCreate(savedInstanceState);
        ^
  symbol:   variable super
  location: class PocActivity
PocActivity.java:36: error: package R does not exist
        setContentView(R.layout.main);
                        ^
PocActivity.java:38: error: cannot find symbol
	txtView = (TextView) findViewById(R.id.txtView);
	           ^
  symbol:   class TextView
  location: class PocActivity
PocActivity.java:38: error: package R does not exist
	txtView = (TextView) findViewById(R.id.txtView);
	                                   ^
PocActivity.java:39: error: cannot find symbol
	Button goBtn = (Button) findViewById(R.id.goBtn);
	^
  symbol:   class Button
  location: class PocActivity
PocActivity.java:39: error: cannot find symbol
	Button goBtn = (Button) findViewById(R.id.goBtn);
	                ^
  symbol:   class Button
  location: class PocActivity
PocActivity.java:39: error: package R does not exist
	Button goBtn = (Button) findViewById(R.id.goBtn);
	                                      ^
PocActivity.java:40: error: cannot find symbol
	mImage = (ImageView)findViewById(R.id.image);
	          ^
  symbol:   class ImageView
  location: class PocActivity
PocActivity.java:40: error: package R does not exist
	mImage = (ImageView)findViewById(R.id.image);
	                                  ^
PocActivity.java:44: error: cannot find symbol
	    InputStream ims = getAssets().open("anakin.png");
	                      ^
  symbol:   method getAssets()
  location: class PocActivity
PocActivity.java:45: error: cannot find symbol
	    Drawable d = Drawable.createFromStream(ims, null);
	    ^
  symbol:   class Drawable
  location: class PocActivity
PocActivity.java:45: error: cannot find symbol
	    Drawable d = Drawable.createFromStream(ims, null);
	                 ^
  symbol:   variable Drawable
  location: class PocActivity
PocActivity.java:47: error: cannot find symbol
	    Log.i("PoC", "Image displayed");
	    ^
  symbol:   variable Log
  location: class PocActivity
PocActivity.java:50: error: cannot find symbol
	    Log.e("PoC", "Image display exception caught: "+exp.toString());
	    ^
  symbol:   variable Log
  location: class PocActivity
PocActivity.java:56: error: package View does not exist
	goBtn.setOnClickListener(new View.OnClickListener()         {
	                                 ^
PocActivity.java:94: error: cannot find symbol
	AssetManager assetmanager = getAssets();
	^
  symbol:   class AssetManager
  location: class PocActivity
PocActivity.java:94: error: cannot find symbol
	AssetManager assetmanager = getAssets();
	                            ^
  symbol:   method getAssets()
  location: class PocActivity
PocActivity.java:112: error: cannot find symbol
	Intent intent =new Intent(Intent.ACTION_VIEW);
	^
  symbol:   class Intent
  location: class PocActivity
PocActivity.java:112: error: cannot find symbol
	Intent intent =new Intent(Intent.ACTION_VIEW);
	                   ^
  symbol:   class Intent
  location: class PocActivity
PocActivity.java:112: error: cannot find symbol
	Intent intent =new Intent(Intent.ACTION_VIEW);
	                          ^
  symbol:   variable Intent
  location: class PocActivity
PocActivity.java:113: error: cannot find symbol
	Log.i("PoC", "File: "+filename);
	^
  symbol:   variable Log
  location: class PocActivity
PocActivity.java:115: error: cannot find symbol
	intent.setDataAndType(Uri.fromFile(new File(filename)), "application/vnd.android.package-archive");
	                      ^
  symbol:   variable Uri
  location: class PocActivity
PocActivity.java:117: error: cannot find symbol
	intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
	                ^
  symbol:   variable Intent
  location: class PocActivity
PocActivity.java:118: error: cannot find symbol
	Log.i("PoC", "Intent is: "+intent.toString()); 
	^
  symbol:   variable Log
  location: class PocActivity
" 
basically the problem is that it cant find the android repositories. I dont have the original set up, only this java file. How can i run it?
thanks

**Comment by cryptax** [comment]
Have you installed Android Studio or the Android SDK?
Those classes you are missing, android.view, android.widget, are pretty basic ones. Your dev environment is probably not set up correctly.
